lib/Recompile/LinkForTranslation.cpp

```diff
@@ -163,7 +163,8 @@ static CommandList linkingArgs(const model::Binary &Model,
                        "-no-pie",
                        "-o",
                        LinkerOutput.path().str(),
-                       ("-Wl,-z,max-page-size=" + Twine(PageSize)).str(),
+                       "-Wl,-z,noexecstack,-z,max-page-size="
+                         + std::to_string(PageSize),
```

Comment:
From my understanding, when `clang` compiles a `.bc` file if there's no indication on the bits of the stack (from the file) it will give it `RWX` by default. However with `-z noexecstack` the default will be to just give `RW`.
I've tested this (just in case), and indeed the translated executable lost the `X` bit when adding `-z noexecstack`

Reply:
Are you sure? I think the issue is for `revng-qa.compiled` files that start from a `.S` file.
Are you sure the problem is with *translated* binaries?

Reply:
I just checked all `calc-*--static-revng-qa.compiled`:
```
root/share/revng/test/tests/runtime/calc-arm-static-revng-qa.compiled
  GNU_STACK      0x000000 0x00000000 0x00000000 0x00000 0x00000 RWE 0x10
root/share/revng/test/tests/runtime/calc-aarch64-static-revng-qa.compiled
  GNU_STACK      0x0000000000000000 0x0000000000000000 0x0000000000000000
                 0x0000000000000000 0x0000000000000000  RW     0x10
root/share/revng/test/tests/runtime/calc-i386-static-revng-qa.compiled
  GNU_STACK      0x000000 0x00000000 0x00000000 0x00000 0x00000 RW  0x10
root/share/revng/test/tests/runtime/calc-mipsel-static-revng-qa.compiled
  GNU_STACK      0x000000 0x00000000 0x00000000 0x00000 0x00000 RWE 0x10

root/share/revng/test/tests/runtime/calc-mips-static-revng-qa.compiled
  GNU_STACK      0x000000 0x00000000 0x00000000 0x00000 0x00000 RWE 0x10

root/share/revng/test/tests/runtime/calc-x86-64-static-revng-qa.compiled
  GNU_STACK      0x0000000000000000 0x0000000000000000 0x0000000000000000
                 0x0000000000000000 0x0000000000000000  RW     0x10
root/share/revng/test/tests/runtime/calc-s390x-static-revng-qa.compiled
  GNU_STACK      0x0000000000000000 0x0000000000000000 0x0000000000000000
                 0x0000000000000000 0x0000000000000000  RW     0x10
```
The error shows up with the `x86-64` executable, so it's a problem with the recompiled binary
I'm curious to know why arm and mips have RWX and the others not, but that's another set of issues

Reply:
So we have two problems:
1. The translated executable has `+rwx` on a segment for some reason.
2. MIPS and ARM `revng-qa.compiled` binaries are `+rwx` for some other reasons.

Correct?

Reply:
Basically yes, just to clarify
1. *All* translated executables have a RWX stack because in the bitcode file we don't say what permissions has the stack, so the linker throws its hands up, gives RWX (RW if `-z noexecstack`) and complains (if RWX)
2. When compiling `.c` files in `revng-qa` some architectures have a `RW` stack and others have a `RWX` stack, however I'm not sure if it's a architecture thing or not (since when compiling from `.c` the compiler will add RW on its own)

Reply:
1. `touch lol.bc; clang lol.bc -shared -o lol; readelf -l lol`. I don't see that.
2. Mh, OK, let's leave the alone for now then.

Reply:
Dunno, I've had a few tries in reproducing the bug but to no avail.
My hypothesis are either:
* clang injects the `.note.GNU-stack` at some late point, and when we compile we're past that but not in the `lol` case
* QEMU might have some places where the stack is executable and this taints the stack during linking